### PR TITLE
support for geolocation queries

### PR DESCRIPTION
**Pull request by kjaskiewiczz**
Ticket: Men 5912

### CODE REVIEW

File: model/filters.go [hunk 1]
```diff
@@ -59,6 +61,33 @@ type FilterPredicate struct {
...
+}
+
+type BoundingBox struct {
+	TopLeft     GeoPoint `json:"top_left" bson:"top_left"`
+	BottomRight GeoPoint `json:"bottom_right" bson:"bottom_right"`
+}
+
+type GeoPoint struct {
+	Lat float32 `json:"lat" bson:"lat"`
+	Lon float32 `json:"lon" bson:"lon"`
```
Comment (merlin-northern) [inline model/filters.go:88]: another nit, but I am usre we can afford the full names in the identifiers.
```suggestion
	Latitude float32 `json:"lat" bson:"lat"`
	Longitude float32 `json:"lon" bson:"lon"`
```

File: model/query.go [hunk 2]
```diff
@@ -562,6 +594,11 @@ func BuildQuery(params SearchParams) (Query, error) {
 		query = devs.AddTo(query)
 	}
 
+	if params.GeoDistanceFilter != nil || params.GeoBoundingBoxFilter != nil {
+		f := NewGeoFilters(params.GeoDistanceFilter, params.GeoBoundingBoxFilter)
+		query = f.AddTo(query)
+	}
+
```
Comment (merlin-northern) [inline model/query.go:601]: just to make me calm: it is ok that one of them is nil, right?

**Review by merlin-northern — COMMENTED** [review]
looks good, left some comments, of which some are real nitpicks.

File: app/indexer/jobs.go [hunk 3]
```diff
@@ -200,6 +205,39 @@ func (i *indexer) processJobDevice(
 	return device
 }
 
+func extractLocation(
+	attrs inventory.DeviceAttributes,
+) (bool, string) {
+	latIdx := -1
+	lonIdx := -1
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply app/indexer/jobs.go:212]: UndefinedCoordinate suggests it's a coordinate, which it isn't;
I can define undefinedCoordinateIdx if you like

File: model/query.go (hunk 2 above)
Comment (kjaskiewiczz) [reply model/query.go:601]: yes

File: docs/management_api.yml [hunk 4]
```diff
@@ -792,6 +800,65 @@ components:
...
+        lat:
+          type: number
+          format: float
+        lon:
+          type: number
+          format: float
+      required:
+        - lat
+        - lon
+
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply docs/management_api.yml]: done

File: model/filters.go (hunk 1 above)
Comment (kjaskiewiczz) [reply model/filters.go]: as you wish

File: tests/tests/test_management_search.py [hunk 5]
```diff
@@ -292,6 +310,12 @@ def __init__(
                             management_api.models.DeviceAttribute(
                                 name="number", value=0.0, scope="inventory"
                             ),
+                            management_api.models.DeviceAttribute(
+                                name="geo-lat", value="59.94", scope="inventory"
+                            ),
+                            management_api.models.DeviceAttribute(
+                                name="geo-lon", value="10.72", scope="inventory"
+                            ),
```
[the remark this answers is not in the document]
Comment (kjaskiewiczz) [reply tests/tests/test_management_search.py:318]: not sure this fits well the acceptance tests; should I add units for validation?

[the remark this answers is not in the document]
Comment (merlin-northern) [reply tests/tests/test_management_search.py:318]: yes.

**Review by merlin-northern — APPROVED** [review]
looks good to me. thanks for applying all the comments.

**Comment by kjaskiewiczz** [comment]
@mender-test-bot start pipeline --fast

**Pull request merged**